code/modules/mob/living/simple_animal/constructs/constructs.dm

```diff
@@ -91,48 +174,74 @@
 /mob/living/simple_animal/construct/armoured
 	name = "Juggernaut"
 	real_name = "Juggernaut"
+	construct_type = "juggernaut"
 	desc = "A possessed suit of armour driven by the will of the restless dead"
 	icon = 'icons/mob/mob.dmi'
 	icon_state = "behemoth"
 	icon_living = "behemoth"
-	maxHealth = 250
-	health = 250
+	maxHealth = 300
+	health = 300
 	response_harm   = "harmlessly punches"
 	harm_intent_damage = 0
 	melee_damage_lower = 30
-	melee_damage_upper = 30
+	melee_damage_upper = 40
+	attack_armor_pen = 60 //Being punched by a living, floating statue.
 	attacktext = list("smashed their armoured gauntlet into")
+	friendly = list("pats")
 	mob_size = MOB_HUGE
-	speed = 3
+	speed = 2 //Not super fast, but it might catch up to someone in armor who got punched once or twice.
 	environment_smash = 2
 	attack_sound = 'sound/weapons/heavysmash.ogg'
 	status_flags = 0
 	resistance = 10
-	construct_spells = list(/spell/aoe_turf/conjure/forcewall/lesser)
+	construct_spells = list(/spell/aoe_turf/conjure/forcewall/lesser,
+							/spell/targeted/fortify,
+							/spell/targeted/construct_advanced/slam
+							)
+
+	armor = list(
+				"melee" = 70,
+				"bullet" = 30,
+				"laser" = 30,
+				"energy" = 30,
+				"bomb" = 10,
+				"bio" = 100,
+				"rad" = 100)
 
 /mob/living/simple_animal/construct/armoured/Life()
 	weakened = 0
 	..()
 
 /mob/living/simple_animal/construct/armoured/bullet_act(var/obj/item/projectile/P)
-	if(istype(P, /obj/item/projectile/energy) || istype(P, /obj/item/projectile/beam))
-		var/reflectchance = 80 - round(P.damage/3)
-		if(prob(reflectchance))
-			adjustBruteLoss(P.damage * 0.5)
+//	if(istype(P, /obj/item/projectile/energy) || istype(P, /obj/item/projectile/beam)) //If it's going to be slow, it's probably going to need every reflect it can get.
+	var/reflectchance = 80 - round(P.damage/3)
+	if(prob(reflectchance))
+		var/damage_mod = rand(2,4)
+		if(!(istype(P, /obj/item/projectile/energy) || istype(P, /obj/item/projectile/beam)))
+			visible_message("<span class='danger'>The [P.name] bounces off of [src]'s shell!</span>", \
+						"<span class='userdanger'>The [P.name] bounces off of [src]'s shell!</span>")
+			new /obj/item/weapon/material/shard/shrapnel(src.loc)
+			var/incoming_brute = (round(P.damage / damage_mod) - (round((P.damage / damage_mod) * 0.3))) //Calculates incoming damage with respect to standard resistance.
+			adjustBruteLoss(incoming_brute)
```

Comment:
Damage type should be dependent on what the projectile's damage type is, and default to this if it's like, tox or something they can't take.

Reply:
This also applies to the else clause below